hudi-client/hudi-client-common/src/test/java/org/apache/hudi/io/TestSecondaryIndexStreamingTracker.java

```diff
@@ -0,0 +1,709 @@
+    // Create a HoodieKey
+    String recordKey = "test-record-key-002";
+    String partitionPath = "2023/01/02";
+    HoodieKey hoodieKey = new HoodieKey(recordKey, partitionPath);
+    
+    // Create old record
+    GenericRecord oldAvroRecord = new GenericData.Record(schema);
+    oldAvroRecord.put("_hoodie_record_key", recordKey);
+    oldAvroRecord.put("_hoodie_partition_path", partitionPath);
+    oldAvroRecord.put("_hoodie_commit_time", "20231201120000");
+    oldAvroRecord.put("_hoodie_commit_seqno", "001");
+    oldAvroRecord.put("_hoodie_file_name", "test-file.parquet");
+    oldAvroRecord.put("id", recordKey);
+    oldAvroRecord.put("name", "Old User");
+    oldAvroRecord.put("fare", 100.5); // Same fare value
+    oldAvroRecord.put("timestamp", System.currentTimeMillis());
+    
+    HoodieAvroPayload oldPayload = new HoodieAvroPayload(Option.of(oldAvroRecord));
+    HoodieRecord oldRecord = new HoodieAvroRecord<>(hoodieKey, oldPayload);
+    
+    // Create new record with same fare value
+    GenericRecord newAvroRecord = new GenericData.Record(schema);
+    newAvroRecord.put("_hoodie_record_key", recordKey);
+    newAvroRecord.put("_hoodie_partition_path", partitionPath);
+    newAvroRecord.put("_hoodie_commit_time", "20231201130000");
+    newAvroRecord.put("_hoodie_commit_seqno", "002");
+    newAvroRecord.put("_hoodie_file_name", "test-file.parquet");
+    newAvroRecord.put("id", recordKey);
+    newAvroRecord.put("name", "Updated User");
+    newAvroRecord.put("fare", 100.5); // Same fare value
+    newAvroRecord.put("timestamp", System.currentTimeMillis());
+    
+    HoodieAvroPayload newPayload = new HoodieAvroPayload(Option.of(newAvroRecord));
+    HoodieRecord newRecord = new HoodieAvroRecord<>(hoodieKey, newPayload);
+    
+    // Create secondary index definition
+    HoodieIndexDefinition indexDef = HoodieIndexDefinition.newBuilder()
+        .withIndexName(MetadataPartitionType.SECONDARY_INDEX.getPartitionPath() + "_fare_index")
+        .withIndexType("SECONDARY_INDEX")
+        .withSourceFields(Collections.singletonList("fare"))
+        .build();
+    
+    // Test case: Update with same secondary key (shouldUpdate = false)
+    SecondaryIndexStreamingTracker.trackSecondaryIndexStats(
+        hoodieKey,
+        Option.of(newRecord),
+        oldRecord,
+        false,
+        writeStatus,
+        schema,
+        () -> schema,
+        Arrays.asList(indexDef),
+        Option.empty(),
+        config
+    );
+    
+    // No secondary index updates should be recorded
+    assertEquals(0, writeStatus.getIndexStats().getSecondaryIndexStats().size());
+  }
+
+  /**
+   * Given: An old record and a new record with different secondary key values
+   * When: trackSecondaryIndexStats processes the update
+   * Then: Two entries should be recorded - delete old value (isDeleted=true) and insert new value (isDeleted=false)
+   */
+  @Test
+  public void testTrackSecondaryIndexStats_UpdateWithDifferentSecondaryKey() {
```

Comment:
can we add null value handing for all these cases. all these are UTs right. should not take up much run time